keyboards/nullbitsco/tidbit/keymaps/oled/keymap.c

```diff
@@ -0,0 +1,149 @@
+/* Copyright 2021 Jay Greco
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+bool numlock_set = false;
+
+enum layers {
+    _BASE = 0,
+    _VIA1,
+    _VIA2,
+    _VIA3
+};
+
+enum custom_keycodes {
+    PROG = SAFE_RANGE,
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [_BASE] = LAYOUT(
+                           KC_PSLS, KC_PAST, KC_PMNS, 
+  KC_VOLD, KC_VOLU, KC_P7, KC_P8,   KC_P9,   KC_PPLS, 
+  KC_TRNS, KC_TRNS, KC_P4, KC_P5,   KC_P6,   KC_PPLS, 
+  KC_TRNS, KC_TRNS, KC_P1, KC_P2,   KC_P3,   KC_PENT, 
+  KC_TRNS, KC_TRNS, KC_P0, KC_P0,   KC_PDOT, KC_PENT  
+  ),
+
+  [_VIA1] = LAYOUT(
+                 ___, ___, ___, 
+  ___, ___, ___, ___, ___, ___, 
+  ___, ___, ___, ___, ___, ___, 
+  ___, ___, ___, ___, ___, ___, 
+  ___, ___, ___, ___, ___, ___  
+  ),
+
+  [_VIA2] = LAYOUT(
+                 ___, ___, ___, 
+  ___, ___, ___, ___, ___, ___, 
+  ___, ___, ___, ___, ___, ___, 
+  ___, ___, ___, ___, ___, ___, 
+  ___, ___, ___, ___, ___, ___
+  ),
+
+  [_VIA3] = LAYOUT(
+                 ___, ___, ___, 
+  ___, ___, ___, ___, ___, ___, 
+  ___, ___, ___, ___, ___, ___, 
+  ___, ___, ___, ___, ___, ___, 
+  ___, ___, ___, ___, ___, ___
+  ),
+};
+
+
+#ifdef OLED_ENABLE
```

Comment:
please move the oled code to the keyboard c file, and use the oled_task_kb instead

Reply:
This was the only change I wasn't sure about. There is more than one keymap with OLED functionality -- would moving the OLED code to the keyboard file mean it applies to all keymaps with the OLED enabled?

Reply:
Moving the code to the keyboard's c file allows any of the keymaps to have that set as the default, but can allow for the display to be overridden by users.   It also allows the board to have a default config for the QMK Configurator.

Reply:
Ah, that makes sense. Changed in the latest revision. Thanks!